src/Evolution/DiscontinuousGalerkin/Initialization/Mortars.cpp

```diff
@@ -0,0 +1,100 @@
+
+namespace evolution::dg::Initialization {
+template <size_t Dim, bool AddFluxBoundaryConditionMortars>
+auto Mortars<Dim, AddFluxBoundaryConditionMortars>::apply_impl(
+    const std::vector<std::array<size_t, Dim>>& initial_extents,
+    const Element<Dim>& element, const TimeStepId& next_temporal_id,
+    const std::unordered_map<Direction<Dim>, Mesh<Dim - 1>>& interface_meshes,
+    const std::unordered_map<Direction<Dim>, Mesh<Dim - 1>>&
+        boundary_meshes) noexcept
+    -> std::tuple<MortarMap<evolution::dg::MortarData<Dim>>,
+                  MortarMap<Mesh<Dim - 1>>,
+                  MortarMap<std::array<Spectral::MortarSize, Dim - 1>>,
+                  MortarMap<TimeStepId>> {
+  static constexpr size_t dim = Dim;
+
+  MortarMap<evolution::dg::MortarData<Dim>> mortar_data{};
+  MortarMap<Mesh<Dim - 1>> mortar_meshes{};
+  MortarMap<std::array<Spectral::MortarSize, Dim - 1>> mortar_sizes{};
+  for (const auto& [direction, neighbors] : element.neighbors()) {
+    for (const auto& neighbor : neighbors) {
+      const auto mortar_id = std::make_pair(direction, neighbor);
+      mortar_data[mortar_id];  // Default initialize data
+      mortar_meshes.emplace(
+          mortar_id, ::dg::mortar_mesh(
+                         interface_meshes.at(direction),
+                         ::domain::Initialization::create_initial_mesh(
+                             initial_extents, neighbor, neighbors.orientation())
+                             .slice_away(direction.dimension())));
+      mortar_sizes.emplace(
+          mortar_id,
+          ::dg::mortar_size(element.id(), neighbor, direction.dimension(),
+                            neighbors.orientation()));
+    }
+  }
+
+  // We will need to update the AddFluxBoundaryConditionMortars to check the
+  // actual boundary condition being imposed.
+  if constexpr (AddFluxBoundaryConditionMortars) {
+    for (const auto& direction : element.external_boundaries()) {
+      const auto mortar_id =
+          std::make_pair(direction, ElementId<dim>::external_boundary_id());
+      mortar_data[mortar_id];  // Default initialize data
+      mortar_meshes.emplace(mortar_id, boundary_meshes.at(direction));
+      mortar_sizes.emplace(mortar_id,
+                           make_array<dim - 1>(Spectral::MortarSize::Full));
+    }
+  } else {
+    (void)boundary_meshes;
+  }
+
+  MortarMap<TimeStepId> mortar_next_temporal_ids{};
+  // Since no communication needs to happen for boundary conditions
+  // the temporal id is not advanced on the boundary, so we only need to
+  // initialize it on internal boundaries
+  for (const auto& direction_and_neighbors : element.neighbors()) {
+    const auto& direction = direction_and_neighbors.first;
+    const auto& neighbors = direction_and_neighbors.second;
+    for (const auto& neighbor : neighbors) {
+      const auto mortar_id = std::make_pair(direction, neighbor);
+      mortar_next_temporal_ids.insert({mortar_id, next_temporal_id});
```

Comment:
Why is this in a separate loop over neighbors? Why not perform this insertion in the earlier loop, around line 54?

If it has to stay down here, then
- add a comment saying why
- use structured bindings `[direction, neighbors] : element.neighbors()` for consistency with the prior loops